File: 1.1/src/cz/mg/vulkan/PFNvkDestroyPipeline.java
package cz.mg.vulkan;

public class PFNvkDestroyPipeline extends VkFunctionPointer {
    public PFNvkDestroyPipeline() {
    }

    protected PFNvkDestroyPipeline(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkDestroyPipeline(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkDestroyPipeline(long value) {
        setValue(value);
    }

    public PFNvkDestroyPipeline(VkInstance instance) {
        super(instance, new VkString("vkDestroyPipeline"));
    }

    public void call(VkDevice device, VkPipeline pipeline, VkAllocationCallbacks pAllocator){
        callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), pipeline != null ? pipeline.getVkAddress() : VkPointer.getNullAddressNative(), pAllocator != null ? pAllocator.getVkAddress() : VkPointer.NULL);
    }

    protected static native void callNative(long vkaddress, long device, long pipeline, long pAllocator);

}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkDestroyPipeline.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkDestroyPipeline_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pipeline, jlong pAllocator){
    (void)env;
    (void)jc;
    PFN_vkDestroyPipeline f = (PFN_vkDestroyPipeline)jniLongToFunctionPointer(address);
    f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkPipeline*)jniLongToPointer(pipeline)),
        ((VkAllocationCallbacks*)jniLongToPointer(pAllocator))
    );
}

